packages/react/fast-components-react-base/src/stack-panel/stack-panel.tsx

```diff
@@ -0,0 +1,570 @@
+     */
+    private getBufferLength = (): number => {
+        if (!isNil(this.props.preloadBufferCount) && this.props.preloadBufferCount >= 0) {
+            return Math.floor(this.props.preloadBufferCount);
+        }
+
+        return 1;
+    };
+
+    /**
+     *  Handle scroll events
+     */
+    private onScrollCapture = (): void => {
+        this.lastRecordedScroll = this.getScrollPosition();
+        clearTimeout(this.scrollLayoutUpdateTimer);
+        if (this.props.scrollLayoutUpdateDelay !== 0) {
+            this.scrollLayoutUpdateTimer = window.setTimeout((): void => {
+                this.updateLayout();
+            }, this.props.scrollLayoutUpdateDelay);
+        } else {
+            this.updateLayout();
+        }
+        this.reportScrollChange();
+    };
+
+    /**
+     *  Handle viewport resize events
+     */
+    private handleResize = (): void => {
+        this.viewportSpan = this.getViewportSpan();
+        this.reportScrollChange();
+        this.updateLayout();
+    };
+
+    /**
+     *  report scroll changes to callback
+     */
+    private reportScrollChange = (): void => {
+        if (isFunction(this.props.onScrollChange)) {
+            this.props.onScrollChange(
+                this.lastRecordedScroll,
+                this.maxScroll,
+                this.viewportSpan
+            );
+        }
+    };
+
+    /**
+     *  updates the direction in state if necessary
+     */
+    private updateDirection = (): void => {
+        const newDirection: Direction = this.getDirection();
+        if (newDirection !== this.state.direction) {
+            this.setState({
+                direction: newDirection,
+            });
+        }
+    };
+
+    /**
+     *  gets the current direction
+     */
+    private getDirection = (): Direction => {
+        if (this.rootElement.current === null) {
+            return Direction.ltr;
+        }
+
+        const closest: Element = this.rootElement.current.closest(
+            `[${StackPanel.DirectionAttributeName}]`
+        );
+
+        return closest === null ||
+            closest.getAttribute(StackPanel.DirectionAttributeName) === Direction.ltr
+            ? Direction.ltr
+            : Direction.rtl;
+    };
+
+    /**
+     *  Gets the scroll position and accounts for direction
+     */
+    private getScrollPosition = (): number => {
+        if (isNil(this.rootElement.current)) {
+            return 0;
+        }
+
+        let scrollPos: number = 0;
+
+        if (this.props.orientation === Orientation.vertical) {
+            scrollPos = this.rootElement.current.scrollTop;
+        } else {
+            scrollPos = RtlScrollConverter.getScrollLeft(
+                this.rootElement.current,
+                this.state.direction
+            );
+            scrollPos = this.state.direction === Direction.rtl ? -scrollPos : scrollPos;
+        }
+
+        return scrollPos;
```

Comment:
nit, just a suggestion but you can avoid declaring a variable and do:
```ts
return this.props.orientation === Orientation.vertical
    ? this.rootElement.current.scrollTop
    : this.state.direction === Direction.rtl
    ? RtlScrollConverter.getScrollLeft(
        this.rootElement.current,
        this.state.direction
    )
    : -(
        RtlScrollConverter.getScrollLeft(
            this.rootElement.current,
            this.state.direction
        )
    );

```

Reply:
yep